static/app/components/dashboards/widgetQueriesForm.tsx

```diff
@@ -45,11 +45,21 @@ type Props = {
   fieldOptions: ReturnType<typeof generateFieldOptions>;
 };
 
+type State = {
+  blurTimeout?: ReturnType<typeof setTimeout>;
```

Comment:
Should this be in state, or just a property on the instance?

Reply:
I'm thinking this has to be in state, effectively I'm using this state to track if the component has fired a `handleFieldChange` from the SearchBar `onSearch` handler to prevent the `onBlur` handler from fire at the same time. `SmartSearchBar` seems to fire both events when selecting something from the autocomplete dropdown. This might not be the most graceful way of solving this issue but I'm also unsure if this is a bug or intended behaviour of `SmartSearchBar`. Let me know what you think! Thanks Evan!

Reply:
I don't think it actually needs to produce a re-render though does it?

Reply:
Ah I think I get what you're saying. Made an update and moved it out of state!